scripts/nbconverted/integrate-msigdb.py

```diff
@@ -0,0 +1,398 @@
+
+
+# ## Add gene set nodes and associated genes as edges
+# 
+# Add each MSigDB collection as distinct nodes with a `participates` edge for representative gene sets and corresponding membership.
+
+# In[13]:
+
+
+def add_msigdb_node_to_graph(current_graph, collection_file, collection_kind,
+                             collection_source, gene_list, min_geneset_size=4,
+                             max_geneset_size=1000, license='CC BY 4.0'):
+    """
+    Add nodes and edges to current graph based on geneset memembership of collection
+    
+    Arguments:
+    current_graph - a hetnet object to add node-edge info to
+    collection_file - location of msigdb file
+    collection_kind - the kind of node already initialized in the graph
+    collection_source - alternative ID for collection
+    gene_list - a list of genes to consider when building the graph
+    min_geneset_size - filter out a given gene set if it has fewer genes
+    max_geneset_size - filter out a given gene set if it has more genes
+    license - given license associated with node
+    
+    Output:
+    Adds to current graph; Returns the amount of filtered genesets
+    """
+    
+    # Build meta data dictionary to store node info
+    meta_data = {'license': license, 'source': collection_source}
+    
+    # Open the .gmt file and process each geneset
+    filtered_genesets = []
+    with open(collection_file, 'r') as collection_fh:
+        collection_reader = csv.reader(collection_fh, delimiter='\t')
+
+        for row in collection_reader:
+            # Get geneset and and metadata info
+            geneset_name = row[0]
+            meta_data['url'] = row[1]
+            
+            # Process geneset membership
+            genes = row[2:]
+
+            # Filter geneset if its too big or small
+            if min_geneset_size > len(genes) or len(genes) > max_geneset_size:
+                filtered_genesets.append(geneset_name)
+                continue
+                
+            # Add the genesetname as a node (based on collection) to the graph
+            current_graph.add_node(kind=collection_kind, identifier=geneset_name, data=meta_data)
+
+            # Loop through all genes and add to the graph it should be considered
+            for gene in genes:
+                if gene in gene_list:
+                    source_id = ('Gene', gene)
+                    target_id = (collection_kind, geneset_name)
+                    
+                    edge_data = meta_data.copy()
+                    edge_data['unbiased'] = False
```

Comment:
No need for you to set this probably.

Reply:
This was the line I had a question about

Reply:
will remove